### PR TITLE
Gardener tested on PPC64LE with minikube single node deployment

### PR DESCRIPTION
**Pull request by ezeeyahoo**
I would request community to review through my progress.
https://github.com/ezeeyahoo/gardener/commits/master

Overview of changes made:-
25a3171 - Dockerfile (cloud images cannot be used as they are build for amd64)
08289a8 - hack/local-development/dev-setup
a186a5f - hack/install.sh (GOARCH= and GOOS= modified)

Docker base images created successfully

Previous thread: #2986

### CODE REVIEW

**Comment by gardener-robot-ci-1** [comment]
Thank you @ezeeyahoo for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: Dockerfile [hunk 1]
```diff
@@ -9,7 +9,7 @@ ARG EFFECTIVE_VERSION
 RUN make install EFFECTIVE_VERSION=$EFFECTIVE_VERSION
 
 ############# base
-FROM eu.gcr.io/gardener-project/3rd/alpine:3.12.3 AS base
+FROM alpine:3.12.3 AS base
```
Comment (rfranzke) [inline Dockerfile]: Can you revert these changes again to use the GCR for the images? :)

Comment (ezeeyahoo) [reply Dockerfile]: We cannot use GCR image for other ports as currently available images are build on AMD64 only, and therefore binaries are incompatible.
Shall I keep my changes in separete new Dockerfile?
or 
Shall I request to push images for PPC too? If so How can I request?

Comment (timuthy) [reply Dockerfile]: Another alternative is to use an ARG e.g., `ARG BASE` where `eu.gcr.io/gardener-project/3rd/` is the default.

https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact.

Comment (ezeeyahoo) [reply Dockerfile]: Please review now.

File: Dockerfile [hunk 2]
```diff
@@ -1,5 +1,6 @@
 #############      builder       #############
-FROM eu.gcr.io/gardener-project/3rd/golang:1.15.5 AS builder
+ARG GCR_PULL_URL
```
Comment (rfranzke) [inline Dockerfile]: /invite @AndreasBurger 

I guess the `.ci/pipeline_definitions` file must be enhanced as well so that the CI/CD system will use the GCR_PULL_URL like before as well.

Comment (timuthy) [reply Dockerfile]: I'd suggest to use a default value directly in the Dockerfile i.e., `ARG GCR_PULL_URL=eu.gcr.io/gardener-project/3rd`

**Comment by ezeeyahoo** [comment]
cc: @gerrith3

File: Dockerfile (hunk 2 above)
Comment (ezeeyahoo) [reply Dockerfile]: @timuthy and pass blank/empty string while 'make'ing for PPC and other ports going forward, right?

Comment (timuthy) [reply Dockerfile]: Yes, although you need to twist this logic in the Makefile again, so that we reach the following:

`make docker-images` --> `... @docker build --build-arg EFFECTIVE_VERSION=$(EFFECTIVE_VERSION) -t $(APISERVER_IMAGE_REPOSITORY):$(EFFECTIVE_VERSION)          -t $(APISERVER_IMAGE_REPOSITORY):latest          -f Dockerfile --target apiserver . ...`

--> uses default value provided in Dockerfile (no build arg)

and 

`make docker-images GCR_PULL_URL=''` --> `... @docker build --build-arg EFFECTIVE_VERSION=$(EFFECTIVE_VERSION) --build-arg GCR_PULL_URL='' -t $(APISERVER_IMAGE_REPOSITORY):$(EFFECTIVE_VERSION)          -t $(APISERVER_IMAGE_REPOSITORY):latest          -f Dockerfile --target apiserver . ...`

--> uses value provided from outside

Feel free to choose a more elegant way but at the moment this is the only option I can think of.

**Review by rfranzke — DISMISSED** [review]
/lgtm
/merge squash
/reviewed ok-to-test

**Review by timuthy — DISMISSED** [review]

**Comment by ezeeyahoo** [comment]
> /lgtm
> /merge squash
> /reviewed ok-to-test

I tried fixup but accidentally something wrong has happened.

**Comment by ezeeyahoo** [comment]
Okay so I have repushed from another clone and fixed up to the point it is possible. Please review.

File: Dockerfile [hunk 3]
```diff
@@ -1,5 +1,6 @@
 #############      builder       #############
-FROM eu.gcr.io/gardener-project/3rd/golang:1.15.5 AS builder
+ARG GCR_PULL_URL=eu.gcr.io/gardener-project/3rd/
+FROM ${GCR_PULL_URL}golang:1.15.5 AS builder
```
Comment (ialidzhikov) [inline Dockerfile]: Can you explain why this is required?

Comment (timuthy) [reply Dockerfile]: https://github.com/gardener/gardener/pull/3544#discussion_r575993840

Comment (ialidzhikov) [reply Dockerfile]: So `golang:1.15.8` is multi-arch and `eu.gcr.io/gardener-project/3rd/golang:1.15.8` is not? Why we don't fix our images instead of introducing similar workarounds?

Comment (vpnachev) [reply Dockerfile]: @ccwienk has spent a lot of effort to improve the CI/CD and we also have dockerhub account, so we should not see the rate limits and can try to switch back to the official `golang:<version>` image.

Comment (timuthy) [reply Dockerfile]: @vpnachev happy to do that. Can you verify this?

Comment (vpnachev) [reply Dockerfile]: > Can you verify this?

I have migrated one pipeline back to dockerhub  and shortly after hit the rate limits :cry:

Comment (ialidzhikov) [reply Dockerfile]: I added a card about it https://github.com/orgs/gardener/projects/6#card-55193026.

**Review by timuthy — APPROVED** [review]

**Comment by rfranzke** [comment]
@ezeeyahoo can you please rebase?

**Comment by ezeeyahoo** [comment]
> @ezeeyahoo can you please rebase?

I have done it to the point I think it was possible from 12 to 7. Further squashing and force pushing resulting into repush of previous merges in my name. 

> 
> Commits on Jan 13, 2021
> 
>     Merge branch 'master' of https://github.com/gardener/gardener
>     @ezeeyahoo
>     ezeeyahoo committed on 13 Jan
> 
> Commits on Feb 17, 2021
> 
>     This commit is working on PPC64LE under single node dep config using … 
> 
> @ezeeyahoo
> ezeeyahoo committed 2 days ago
> 
>

**Comment by rfranzke** [comment]
Hm, I don't get it 👀 There are still conflicts preventing this PR from being merged.
